core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -528,20 +528,26 @@ private static DefaultFunctionResolver truncate() {
    * Definition of multiply(x, y) function.
    * Returns the number x multiplied by number y
    * The supported signature of multiply function is
-   * (x: INTEGER/LONG/FLOAT/DOUBLE, y: INTEGER/LONG/FLOAT/DOUBLE)
+   * (x: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE, y: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE)
    * -> wider type between types of x and y
    */
   private static DefaultFunctionResolver multiplyFunction() {
     return FunctionDSL.define(BuiltinFunctionName.MULTIPLYFUNCTION.getName(),
             FunctionDSL.impl(
                     FunctionDSL.nullMissingHandling(
-                            (x, y) -> new ExprIntegerValue(Math.multiplyExact(x.integerValue(),
-                                    y.integerValue()))),
+                            (x, y) -> new ExprByteValue((x.byteValue() * y.byteValue()))),
+                    BYTE, BYTE, BYTE),
+            FunctionDSL.impl(
+                    FunctionDSL.nullMissingHandling(
+                            (x, y) -> new ExprShortValue(x.shortValue() * y.shortValue())),
+                    SHORT, SHORT, SHORT),
```

Comment:
multiplication of 2 `BYTE`s could not fit in `BYTE` and so on. Consider returning bigger value type where it is possible.
Fortunately, all numeric ExprValues accept and store `Number` type, so no data lost. It is nice to have a test for that for `multiply` and for `*` both.

Reply:
When trying to conduct any math operator that causes an overflow the OS SQL CLI returns an overflow exception, currently my implementation just loops around the number. Should I keep my function like the other arithmetic functions and return an error, or should I make it work around overflows?

Reply:
Currently `MULTIPLY()` acts exactly like using the `*` symbol, including throwing an overflow if the number is too big to handle, e.g.
`SELECT MULTIPLY(2000000000, 2);` and `SELECT 2000000000 * 2;`
will both return: 
`TransportError(503, 'ArithmeticException', {'error': {'reason': 'There was internal problem at backend', 'details': 'integer overflow', 'type': 'ArithmeticException'}, 'status': 503})`

Should I keep the functionality as it is or change it, and if so should I change it for both `MULTIPLY()` and `*`?